C/fibonacci/fibonacci.c

```diff
@@ -1,15 +1,33 @@
 #include <stdio.h>
+#include <stdlib.h>
+
+#define DEFAULT_N 10
 
 int FibonacciSequence(int num) {
     if(num <= 1) {
         return num ;
     }
     return FibonacciSequence(num-1) + FibonacciSequence(num-2);
 }
-int main(){
-    printf("Enter the Number\n");
-    int n ;
-    scanf("%d",&n);
 
-    printf("Fibonacci Sequence term at %d is %d " , n , FibonacciSequence(n));
+int main(int argc, char* argv[]){
+	int n = DEFAULT_N;
+    if (argc > 1) {
+    	n = atoi(argv[1]);
+    } else {
+    	printf("Which Fibonacci index to find? ");
+    	int matched = scanf("%d", &n);
+    	if (matched < 1) {
```

Comment:
0 is a valid index
If it's negative it should return an error

Reply:
0 is allowed, and it goes to the default if negative

Reply:
No, it does not
```
$ nix build github:rjzak/codex/fix_c_cpp_go_ruby#fibonacci-c
$ echo 0 | enarx run --wasmcfgfile ./result/Enarx.toml ./result/main.wasm 
Which Fibonacci index to find? Fibonacci sequence number at index 0 is 0
$ echo -1 | enarx run --wasmcfgfile ./result/Enarx.toml ./result/main.wasm
Which Fibonacci index to find? Fibonacci sequence number at index -1 is -1
$ echo -20 | enarx run --wasmcfgfile ./result/Enarx.toml ./result/main.wasm
Which Fibonacci index to find? Fibonacci sequence number at index -20 is -20
```

From `man 3 scanf`:
> On success, these functions return the number of input items successfully matched and assigned; this can be fewer than provided for, or even zero, in the event of an early matching failure.